.circleci/config.yml

```diff
@@ -1,55 +1,255 @@
+          echo "deb http://aptly.greenzie.com/greenzie-<< parameters.server_type >> ${UBUNTU_DISTRO} main" >> /etc/apt/sources.list
+          cat /etc/apt/sources.list
+
+checkout_codebase: &checkout_codebase
+  checkout:
+    path: /root/catkin_ws/src/fixposition_driver
+
+update_environment: &update_environment
+  run:
+    name: "Update Docker Image"
+    command: |
+      apt-get update
+      apt-get upgrade -y
+      apt-get install python3-hurry.filesize
+
+preparing_ros_workspace: &preparing_ros_workspace
+  - run:
+      name: "Prepping ROS workspace"
+      command: |
+        source /opt/ros/${ROS_DISTRO}/setup.bash
+        mkdir -p src
+        catkin_make -j 1
+
+update_git_submodules: &update_git_submodules
+  - run:
+      name: "Update Git Submodules"
+      command: |
+        cd src/fixposition_driver && git submodule init
+        git submodule update --recursive --remote
+
+source_ros_setup: &source_ros_setup
+  - run:
+      name: "Install ROSDEP Dependencies"
+      command: |
+        source /opt/ros/${ROS_DISTRO}/setup.sh
+
+source_greenzie_setup: &source_greenzie_setup
+  - run:
+      name: "Install Greenzie ROSDEP Dependencies"
+      command: |
+        bash "${HOME}/catkin_ws/src/fixposition_driver/configurations/install_greenzie_rosdep_source.sh"
+        rosdep update
+        rosdep install -iy --from-paths /root/catkin_ws/src --rosdistro=${ROS_DISTRO}
+
+arm64_source_greenzie_setup: &arm64_source_greenzie_setup
+  - run:
+      name: "Install Greenzie ROSDEP Dependencies"
+      command: |
+        mv "${HOME}/catkin_ws/src/fixposition_driver/configurations/10-greenzie-base-arm64.yaml" "${HOME}/catkin_ws/src/fixposition_driver/configurations/10-greenzie-base.yaml"
+        bash "${HOME}/catkin_ws/src/fixposition_driver/configurations/install_greenzie_rosdep_source.sh"
+        rosdep update
+        rosdep install -iy --from-paths /root/catkin_ws/src --rosdistro=${ROS_DISTRO}
+
+
+########################
+######### STEPS ########
+########################
+ros_env_prep_steps: &ros_env_prep_steps
+  # This 'when' is hack which allows us to concatenate list of steps
+  # We cannot combine lists, so this is the best solution I have
+  # See: https://circleci.com/docs/introduction-to-yaml-configurations/#anchors-and-aliases
+  when:
+    condition:
+      equal: [ "true", "true" ]
+    steps:
+      - when:
+          condition:
+            not:
+              matches:
+                pattern: "^(noetic|release-candidate-v.*)$"
+                value: << pipeline.git.branch >>
+          steps:
+            - add_package_server:
+                server_type: "experimental"
+      - when:
+          condition:
+            matches:
+              pattern: "^(release-candidate-v.*)$"
+              value: << pipeline.git.branch >>
+          steps:
+            - add_package_server:
+                server_type: "rc"
```

Comment:
We don't go through the `release-candidate` steps with dependency packages. The only rule we've used in the past is to deploy to "experimental" on merge to the `noetic-devel` branch.